lib/backend/k8s/resources/tpr.go

```diff
@@ -0,0 +1,269 @@
+// Copyright (c) 2017 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resources
+
+import (
+	"reflect"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/projectcalico/libcalico-go/lib/backend/model"
+	"github.com/projectcalico/libcalico-go/lib/errors"
+
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	extensions "k8s.io/client-go/pkg/apis/extensions/v1beta1"
+	"k8s.io/client-go/rest"
+)
+
+// Interface required to satisfy use as a Kubernetes Third Party Resource.
+type k8sResource interface {
+	runtime.Object
+	metav1.ObjectMetaAccessor
+}
+
+// Interface required to satisfy use as a Kubernetes Third Party Resource List.
+type k8sList interface {
+	runtime.Object
+	metav1.ListMetaAccessor
+}
+
+// tprHandler implements the ResourceHandler interface and provides a generic
+// mechanism for a 1:1 mapping between a Calico Resource and an equivalent
+// Kubernetes Third Party Resource.
+type tprHandler struct {
```

Comment:
Yup that's one of the things I wanted to discuss w/ you when we do an interface overview.